src/@apollo/client/react/types/ApolloClient__React_Types.res

```diff
@@ -730,44 +725,27 @@ module MutationHookOptions = {
     // export interface MutationHookOptions<TData = any, TVariables = OperationVariables> extends BaseMutationOptions<TData, TVariables> {
     //   mutation?: DocumentNode;
     // }
-    @ocaml.doc("
-     * We use deriving abstract here because option('jsVariables) is not typesafe,
-     * but still needs to be optional since a user has the option of providing
-     * the variables to the results of the hook rather than the hook itself
-     ")
```

Comment:
1. There's a few of these comments left over where the types aren't using `@deriving(abstract)` anymore, I can remove them if you think they aren't relevant anymore
2. `deriving(abstract)` seems to have the same effect as the `@obj` approach, figured I'd keep them all consistent but if there is a preference for one over the other I can change it.

Reply:
I don't know that I prefer one over the other. I'd forgotten we use deriving actually. It does have the bonus of being able to convert in both directions, but in these cases, that's not actually happening. `@obj` seems fine to me. I think the general comment is still relevant. Maybe just switch deriving to `@obj`. I think there was an issue when nesting optionals or something that created an unexpected representation in some cases. Now I wish I'd written a better comment, though!

Thanks for going the extra mile for consistency!